### PR TITLE
Output sequences

### PR DESCRIPTION
**Pull request by yeeplusplus**
Initial solution for #7

### CODE REVIEW

File: shpy [hunk 1]
```diff
@@ -7,22 +7,33 @@ createSpy() {
     fi
 
     # shellcheck disable=SC2039
-    local val output error_output
+    local val output error_output stub_dir outputs_dir last_arg
+    for last_arg; do true; done
```
Comment (codehearts) [inline shpy]: `last_arg` is supposed to always be the spy name, I think this would be more understandable with a name like `spy_name`

File: shpy [hunk 2]
```diff
@@ -7,22 +7,33 @@ createSpy() {
...
 
+    _shpyInit
+
+    stub_dir=$(realpath "$_shpy_spies_dir")
+    outputs_dir=${stub_dir}/outputs/${last_arg}
+    mkdir -p "${outputs_dir}"
```
Comment (codehearts) [inline shpy]: This codebase uses `shpy_ensure_dir` to create the appropriate directory, and doesn't call `realpath`. You should be able to do `shpy_ensure_dir "$_shpy_spies_dir/outputs/$spy_name"` in place of these three lines

File: shpy [hunk 3]
```diff
@@ -192,8 +203,22 @@ _shpySpySetErrorOutput() {
...
     # shellcheck disable=SC2039
-    local output
-    eval output="\$_shpy_${name}_output"
+    local call_index count output output_fixture
+
+    stub_dir=$(realpath "$_shpy_spies_dir")
+    outputs_dir=${stub_dir}/outputs/${name}
+
+    count=$(getSpyCallCount "$name")
+    call_index=$((count - 1))
```
Comment (codehearts) [inline shpy]: If `output_num` in `createSpy` started at 1, this math shouldn't be necessary here

**Comment by codehearts** [comment]
This looks great! Build is failing only because `spellcheck` doesn't like `local` in the tests, I should change my `spellcheck` invocation to ignore that globally.

The readme should be updated to cover these changes, and it would be good to do the same for stderr using `-e`.

**Comment by yeeplusplus** [comment]
What version of shellcheck are you using?  0.4.7 doesn't report any errors

File: shpy (hunk 1 above)
Comment (yeeplusplus) [reply shpy]: Good catch

File: shpy (hunk 3 above)
Comment (yeeplusplus) [reply shpy]: True, but I did it this way to be consistent with `_shpySpySaveArgs`, which saves arguments in separate files starting at 0.

Let me know if you want me to change the outputs to start at 1

File: shpy (hunk 2 above)
Comment (yeeplusplus) [reply shpy]: Fixed

File: shpy (hunk 3 above)
Comment (codehearts) [reply shpy]: In that case, I think it makes sense to leave it at 0

**Comment by codehearts** [comment]
This looks great, thanks for contributing! :tada:

**Pull request merged**